sparktestingbase/testcase.py

```diff
@@ -14,68 +16,103 @@
     For non local mode testing you can either override sparkMaster
-    or set the enviroment property SPARK_MASTER for non-local mode testing."""
-
-    @classmethod
-    def getMaster(cls):
-        return os.getenv('SPARK_MASTER', "local[4]")
+    set the environment property SPARK_MASTER."""
+
+    def get_conf(self):
+        conf = SparkConf()
+        conf.set('spark.sql.session.timeZone', 'UTC')
+        conf.set('spark.sql.shuffle.partitions', '12')
+        return conf
+
+    def create_local_spark_session(self):
+        class_name = self.__class__.__name__
+        spark = SparkSession \
+            .builder \
+            .master("local[4]") \
+            .appName(class_name) \
+            .enableHiveSupport() \
+            .getOrCreate()
+        spark.sparkContext._conf.setAll([("spark.sql.shuffle.partitions", "12"),
+                                         ("spark.sql.session.timeZone", "UTC")])
+        return spark
 
     def setUp(self):
-        """Setup a basic Spark context for testing"""
-        self.sc = SparkContext(self.getMaster())
-        self.sql_context = HiveContext(self.sc)
+        """Setup a basic Spark session for testing"""
+        self.spark = self.create_local_spark_session()
         quiet_py4j()
 
     def tearDown(self):
         """
         Tear down the basic panda spark test case. This stops the running
         context and does a hack to prevent Akka rebinding on the same port.
         """
-        self.sc.stop()
+        self.spark.stop()
         # To avoid Akka rebinding to the same port, since it doesn't unbind
         # immediately on shutdown
-        self.sc._jvm.System.clearProperty("spark.driver.port")
+        self.spark.sparkContext._jvm.System.clearProperty("spark.driver.port")
 
+    def assert_rdd_equals(self, expected, result):
+        return self.compare_rdd(expected, result) == []
 
-class SparkTestingBaseReuse(unittest2.TestCase):
+    def compare_rdd(self, expected, result):
+        expected_keyed = expected \
+            .map(lambda x: (x, 1)) \
+            .reduceByKey(lambda x, y: x + y)
+        result_keyed = result \
+            .map(lambda x: (x, 1)) \
+            .reduceByKey(lambda x, y: x + y)
+        return expected_keyed.cogroup(result_keyed) \
+            .map(lambda x: tuple(map(list, x[1]))) \
+            .filter(lambda x: x[0] != x[1]).take(1)
+
+    def assertRDDEqualsWithOrder(self, expected, result):
+        return self.compare_rdd_with_order(expected, result) == []
 
+    def compare_rdd_with_order(self, expected, result):
+        def index_rdd(rdd):
+            return rdd.zipWithIndex().map(lambda x: (x[1], x[0]))
+
+        index_expected = index_rdd(expected)
+        index_result = index_rdd(result)
+        return index_expected.cogroup(index_result) \
+            .map(lambda x: tuple(map(list, x[1]))) \
+            .filter(lambda x: x[0] != x[1]).take(1)
+
+
+class SparkTestingBaseReuse(unittest2.TestCase):
```

Comment:
Why are the new comparison methods, such as `compare_rdd` only implemented for the `BaseTestCase` and not for `BaseReuse`?

Reply:
I believe we don't need to have 2 classes. SparkSession builder have .getOrCreate() method and it returns current SparkSession if it exist. I removed it and leave only basic class.
Some later I'll add methods to compare dataframes.